server/src/lib/routes/secondfactor/duo-push/Post.ts

```diff
@@ -17,11 +17,13 @@ interface DuoResponse {
   stat: "OK" | "FAIL";
 }
 
-function triggerAuth(username: string, config: DuoPushConfiguration): Promise<DuoResponse> {
+function triggerAuth(username: string, config: DuoPushConfiguration, req: Express.Request): Promise<DuoResponse> {
   return new Promise((resolve, reject) => {
+    const clientIP = req.headers["x-forwarded-for"] || req.connection.remoteAddress;
```

Comment:
Use GetHeader method in server/src/lib/utils.

Reply:
req.ip would be better by the way, it's the single IP of the client, X-Forwarded-For can be a chain of IPs.

Reply:
and req.ip is the IP used in the logger.